### DISCUSSION TITLE
A simple app that has two views (form, window, sub-form i.e.)

### DISCUSSION THREAD

**Discussion by erkanhurnali** [flet-dev/flet#1724] [Q&A]
Dear Flet Developers,

I am a Python programmer who is interested in using your Flet framework to create cross-platform applications. I have been following your tutorials and documentation on your website, but I have encountered a problem that I could not solve.

I wanted to create a simple app that has two views (form, window, sub-form i.e.): a home view and a form view. The app should be able to switch between these views by clicking buttons on each view. I followed the basic app structure that you provided on this page:

https://flet.dev/docs/guides/python/getting-started/

However, when I tried to run the app, I could not see the two views on the screen. I only saw a blank page with the title "Flet counter example". I checked the code several times, but I could not find any errors or mistakes.

I wonder if you could help me with this issue. Is there something wrong with the app method or the view class? How can I create and switch between multiple views using Flet? Do you have any examples or resources that could guide me?

I appreciate your time and attention. Thank you for creating such a useful and innovative framework. I hope to hear from you soon.

Sincerely,

Erkan HÜRNALI
erkanhurnali@gmail.com

**Comment by erkanhurnali** [comment]
Last time I tried a code like this, but without success. So I couldn't reach my goal. :(

import flet as ft
import asyncio

def main(page: ft.Page):
    # Ana sayfa
    home_view = ft.View()
    home_view.controls.append(ft.Text("Merhaba, bu ana sayfa"))
    # Form sayfası
    form_view = ft.View()
    form_view.controls.append(ft.Text("Merhaba, bu form sayfası"))
    # Sayfalar arasında geçiş yapmak için butonlar
    async def go_home(e):
        page.controls.append(ft.Text("Ana sayfaya geçiliyor...")) # Sayfaya bir kontrol ekle
        await page.update_async() # Sayfayı güncelle
        page.view = home_view # Sayfa görünümünü değiştir
        await page.update_async() # Sayfayı güncelle
        await asyncio.sleep(1) # 1 saniye bekle
        page.appbar.title = "Ana Sayfa" # Başlığı güncelle
        await page.update_async() # Sayfayı güncelle
    async def go_form(e):
        page.controls.append(ft.Text("Form sayfasına geçiliyor...")) # Sayfaya bir kontrol ekle
        await page.update_async() # Sayfayı güncelle
        page.view = form_view # Sayfa görünümünü değiştir
        await page.update_async() # Sayfayı güncelle
        await asyncio.sleep(1) # 1 saniye bekle
        page.appbar.title = "Form Sayfası" # Başlığı güncelle
        await page.update_async() # Sayfayı güncelle
    home_button = ft.ElevatedButton("Ana Sayfaya Git", on_click=go_home)
    form_button = ft.TextButton("Form Sayfasına Git", on_click=go_form)
    # Butonları her iki sayfaya da ekle
    home_view.controls.append(form_button)
    form_view.controls.append(home_button)
    # Ana sayfayı göster
    page.view = home_view

# Fonksiyon dışında çalıştır (girintisiz)
ft.app(target=main)

**Comment by rkleivel** [comment]
I have not checked your code, but have you seen this page?
https://flet.dev/docs/guides/python/navigation-and-routing

**Comment by erkanhurnali** [comment]
Thank you for finding it worth commenting. I understand that I did not express my problem fully.

I don't want to cycle through different pages in one window. I want to show-hide a second window inside an application.

So as before; I'm looking for a feature like child-form, sub-form. So I'm nostalgic.😊
